core/src/main/java/io/aiven/klaw/service/SchemaRegistrySyncControllerService.java

```diff
@@ -153,20 +160,49 @@ private List<SchemaSubjectInfoResponse> getSchemasFromDb(
         }
 
         Set<Integer> schemaVersionsInt = new TreeSet<>();
-        schemaVersions.forEach(ver -> schemaVersionsInt.add(Integer.parseInt(ver)));
+        schemaVersions.forEach(ver -> schemaVersionsInt.add(parseIntFromSchemaVersion(ver)));
 
         SchemaSubjectInfoResponse schemaInfo = new SchemaSubjectInfoResponse();
         schemaInfo.setTopic(topic.getTopicname());
         schemaInfo.setSchemaVersions(schemaVersionsInt);
         schemaInfo.setTeamId(topic.getTeamId());
 
         schemaInfoList.add(schemaInfo);
+        topicSchemaVersionsInDb.remove(topic.getTopicname());
       }
     }
+    return schemaInfoList;
+  }
 
+  private List<SchemaSubjectInfoResponse> annotateAnomalousSchemasNotInDB(
+      String schemaEnvId,
+      List<SchemaSubjectInfoResponse> schemaInfoList,
+      Map<String, Set<String>> topicSchemaVersionsInDb,
+      Map<String, SchemaInfoOfTopic> schemaTopicAndVersions) {
+    // Annotate anomalous Schemas
+    Set<Map.Entry<String, Set<String>>> schemaTopicNameToVersionsDB =
+        topicSchemaVersionsInDb.entrySet();
+    for (Map.Entry<String, Set<String>> schemaDetail : schemaTopicNameToVersionsDB) {
+
+      Set<Integer> schemaVersionsInt = new TreeSet<>();
+      schemaDetail.getValue().forEach(ver -> schemaVersionsInt.add(parseIntFromSchemaVersion(ver)));
+      SchemaSubjectInfoResponse schemaInfo = new SchemaSubjectInfoResponse();
+
+      schemaInfo.setTopic(schemaDetail.getKey());
+      schemaInfo.setSchemaVersions(schemaVersionsInt);
+      schemaInfo.setEnvId(schemaEnvId);
+      schemaInfo.setRemarks(annotateSchemasNotInCluster(schemaInfo, schemaTopicAndVersions));
+
+      schemaInfoList.add(schemaInfo);
+    }
     return schemaInfoList;
   }
 
+  private static int parseIntFromSchemaVersion(String ver) {
+    Double dub = Double.parseDouble(ver);
```

Comment:
double dub = Double.parseDouble(ver);
    return (int) dub;

this is even better, as that type may be primitive

Reply:
That actually gives an inconvertible type cannot cast java.lang.Double to int.

Reply:
But if you assign it to double, it should be possible to cast